### DISCUSSION TITLE
Heuristic Tiebreaking in Astar Algorithm

### DISCUSSION THREAD

**Discussion by lior8** [networkx/networkx#5335] [Algorithms]
Currently tiebreaking is done via the insertion counter (i.e., when was the node added to the queue, the smaller the counter, the earlier it was added).
A tie breaker based on the heuristic value and then the insertion counter can lead to much better results as the algorithm will prefer nodes it thinks are closer to the target. Furthermore, in a case when there is 100% accuracy between the heuristic and actual distance, it would provide much better results (which is surprisingly often in some maps and graph types).

Edit: [Here is the relevant PR as well](https://github.com/networkx/networkx/pull/5344).

**Comment by dschult** [comment]
This seems like a very reasonable improvement. And thank you for the PR that I haven't taken a good look at yet.
The only issue from my perspective is whether this is still the Astar Algorithm or a different -- but obviously related -- algorithm.
I think this is a small enough change (an implementation detail) that we could put it in place, flag a change in the output of the function in the Release Notes for v3.0, and not try to maintain an option for the original method. 

What do others think?
